vm_eval.c

```diff
@@ -45,12 +45,20 @@ static VALUE vm_call0_body(rb_execution_context_t* ec, struct rb_calling_info *c
 MJIT_FUNC_EXPORTED VALUE
 rb_vm_call0(rb_execution_context_t *ec, VALUE recv, ID id, int argc, const VALUE *argv, const rb_callable_method_entry_t *me, int kw_splat)
 {
-    struct rb_calling_info calling = { Qundef, recv, argc, kw_splat, };
-    const struct rb_callinfo *ci = vm_ci_new_runtime(id, kw_splat ? VM_CALL_KW_SPLAT : 0, argc, NULL);
-    struct rb_callcache cc_body;
-    const struct rb_callcache *cc = vm_cc_fill(&cc_body, 0, me, vm_call_general);
-    struct rb_call_data cd = { ci, cc, };
-    return vm_call0_body(ec, &calling, &cd, argv);
+    return vm_call0_body(
+        ec,
+        &(struct rb_calling_info) {
+            .block_handler = VM_BLOCK_HANDLER_NONE,
+            .recv = recv,
+            .argc = argc,
+            .kw_splat = kw_splat,
+        },
+        &(struct rb_call_data) {
+            .ci = &VM_CI_ON_STACK(id, kw_splat ? VM_CALL_KW_SPLAT : 0, argc, NULL),
+            .cc = &VM_CC_ON_STACK(Qfalse, vm_call_general, { 0 }, me),
+        },
+        argv
+    );
```

Comment:
@ko1 This is acceptable to me.  Thoughts?
```suggestion
    struct calling_info calling = {
        .block_handler = VM_BLOCK_HANDLER_NONE,
        .recv = recv,
        .argc = argc,
        .kw_splat = kw_splat,
    };
    struct rb_call_data cd = {
        .ci = &VM_CI_ON_STACK(id, kw_splat ? VM_CALL_KW_SPLAT : 0, argc, NULL),
        .cc = &VM_CC_ON_STACK(Qfalse, vm_call_general, { 0 }, me),
    };
    
    return vm_call0_body(ec, &calling, &cd, argv);
```

Reply:
I like this style. But as I mention it as "curious", I want to ask the preference.

Reply:
Preference for compound literals...

Well I think "having no variable" is a good property.  That way a struct is implicitly immutable, because there is no way to reuse it.  If you assign it to a variable, you have to manually make sure it won't be clobbered later.

Also, "having no need to think about variable name" is easier to me.  That's not a big pain, but a pain.

Reply:
Agreed for the reasons.
I feel it is difficult because the deep nesting and difficult to understand the structure.
I'm not sure which should be prioritize...